packages/react-ai/src/components/AIConversation/views/Controls/FieldControl.tsx

```diff
@@ -165,14 +165,42 @@ export const FieldControl: FieldControl = () => {
     if (handleSendMessage) handleSendMessage({ content: submittedContent });
     if (setInput) setInput({ text: '', files: [] });
   };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
```

Comment:
Should these handlers both share the same internal logic after checking for the key?

Reply:
Yeah that was the goal here by moving the submitMessage logic out, but definitely realizing it could be combined more, will make an u pdate

Reply:
Combined it more but couldn't combine the `preventDefault` call because in the keyboard case we always want to `preventDefault` but we do not always want to submit the message, like in the case when the message is empty